src/chess/board.cc

```diff
@@ -759,6 +759,13 @@ Move ChessBoard::GetLegacyMove(Move move) const {
   return move;
 }
 
+Move ChessBoard::GetModernMove(Move move) const {
+  if (our_king_ != E1 || move.from() != E1) return move;
+  if (move == Move(E1, G1) && !our_pieces_.get(G1)) return Move(E1, H1);
```

Comment:
Is the `&& !our_pieces_.get(G1)` part needed?

Reply:
Yes, if there is a rook at G1, then it's already "modern-style" move and should not be converted.